### PR TITLE
Trackalert Performance and Documentation

### PR DESCRIPTION
**Pull request by neilcook**
- Decouple web server thread pool from report thread pool - means that slow reports don't chew up web server threads, as they now run in their own thread pool. Also means that the report API returns OK before the report function is run, which is fine.
- Initial Trackalert README.md
- Man pages for trackalert
- Swagger documentation for trackalert API

### CODE REVIEW

File: common/misc.cc [hunk 1]
```diff
@@ -864,3 +864,12 @@ std::string getDirectoryPath(const std::string& filename)
   else
     return ".";
 }
+
+std::string getFileFromPath(const std::string& filename)
```
Comment (cmouse) [inline common/misc.cc]: http://www.boost.org/doc/libs/1_31_0/libs/filesystem/doc/path.htm

**Comment by neilcook** [comment]
Cool, thx - I’ll use that instead.

File: common/misc.cc [hunk 2]
```diff
@@ -858,18 +859,13 @@ unsigned int pdns_stou(const std::string& str, size_t * idx, int base)
...
 std::string getDirectoryPath(const std::string& filename)
 {
-  size_t found = filename.find_last_of("/\\");
-  if (found != string::npos)
-    return filename.substr(0, found);
-  else
-    return ".";
+  boost::filesystem::path my_path(filename);
+  boost::filesystem::path branch_path = my_path.branch_path();
+  return branch_path.empty() ? "." : branch_path.string();
```
Comment (cmouse) [inline common/misc.cc]: should it use native?

**Comment by neilcook** [comment]
I thought about using native, but:

1) I have no desire for this to work on anything other than POSIX compliant systems
2) Generic works on both Linux and Mac

I’d also be concerned about side-effects with other code, e.g. assuming that g_configDir was in posix format.

Neil

**Pull request merged**